### DISCUSSION TITLE
support for custom table name prefix

### DISCUSSION THREAD

**Discussion by arucil** [SeaQL/sea-orm#1487] [Ideas]
Recently I chose SeaORM when rewriting some modules of a Node.js server in Rust, and I found one important feature that is missing in SeaORM is custom table name prefix. Implementing `EntityName` manually solves the problem, but it also requires implementing many other structs and traits that could have been generated by `DeriveEntityModel`, which is a PITA.

Neither `Entity` and `DatabaseConnection` is suitable for storing a global table name prefix, so IMO a new ORM manager struct is needed, for example, the initialization becomes

```rust
let mut orm: SeaOrm = SeaOrm::new()
   .with_table_name_prefix("my_prefix")
   .with_connection_options(options)
   .build();

let result = my_table::Entity::find().all(&orm).await?;
```

Currently I've added a `table_name_expr` attr to the `DeriveEntityModel` macro in my `sea-orm` fork, and changed the return type of `EntityName::table_name()` and related types from `&str` to `Cow<str>`, so I can write
```rust
#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
#[sea_orm(table_name_prefix = r#"prefix("my_table")"#]
pub struct Model {
    ....
}

fn prefix(table_name: &str) -> String {
    format!("{}_{}", std::env::var("TABLE_PREFIX").unwrap(), table_name)
}
```

It's a bit tedious, but it solves my problem for now.

**Comment by billy1624** [comment]
Hey @arucil, interesting use case! I guess you're working with a legacy DB? Can you share your code by filing a PR?

**Reply by arucil** [reply]
Yes, our database has been in use for years. My sea-orm fork now contains other functionalities that aren't related to custom table name prefix, simply filing a PR from my `master` branch may cause distraction, so I'll post links to commits here (only  2 commits): https://github.com/arucil/sea-orm/commit/10e43e17a856644ea4221242e3ae4785561dbfb5 and https://github.com/arucil/sea-orm/commit/5f7d8a1aa97d19a52d3928584b9085989c4897e9 .
